tools/darwin/Support/Codesign.command

```diff
@@ -2,66 +2,49 @@
 
 set -x
 
-#this is the list of binaries we have to sign for being able to run un-jailbroken
+# This is the list of binaries we have to sign for being able to run un-jailbroken
 LIST_BINARY_EXTENSIONS="dylib so app"
 
-GEN_ENTITLEMENTS="$NATIVEPREFIX/bin/gen_entitlements.py"
 DARWIN_EMBEDDED_ENTITLEMENTS="$XBMC_DEPENDS/share/darwin_embedded_entitlements.xml"
 LDID="$NATIVEPREFIX/bin/ldid"
+TOPSHELF_BINARY="${BUILT_PRODUCTS_DIR}/${EXECUTABLE_FOLDER_PATH}/PlugIns/${TOPSHELF_EXTENSION_NAME}.appex/${TOPSHELF_EXTENSION_NAME}"
```

Comment:
1. I suggest to use `CONTENTS_PATH` defined below
2. correct way to find the right exexcutable is to read `CFBundleExecutable` from Info.plist